SmartDeviceLink/SDLSystemCapabilityManager.m

```diff
@@ -801,6 +792,48 @@ - (void)sdl_hmiStatusNotification:(SDLRPCNotificationNotification *)notification
     self.currentHMILevel = onHMIStatus.hmiLevel;
 }
 
+#pragma mark Utilities
+
+/// Checks if we are already on the serial readWrite queue. If so, the block is added to the queue; if not, the block is dispatched to the readWrite queue.
+/// @discussion Used to ensure atomic access to global properties.
+/// @param block The block to be executed.
+- (void)sdl_runSyncOnQueue:(void (^)(void))block {
+    if (dispatch_get_specific(SDLProcessingQueueName) != nil) {
+        block();
+    } else {
+        dispatch_sync(self.readWriteQueue, block);
+    }
+}
+
+#pragma mark Getters
+
+- (NSMutableDictionary<SDLSystemCapabilityType, NSMutableArray<SDLSystemCapabilityObserver *> *> *)capabilityObservers {
+    __block NSMutableDictionary<SDLSystemCapabilityType, NSMutableArray<SDLSystemCapabilityObserver *> *> *dict = nil;
+    [self sdl_runSyncOnQueue:^{
+        dict = self->_capabilityObservers;
```

Comment:
Why are you using `->` instead of `.`?

Reply:
If you access the variable within its own getter using `self.`, it causes infinite recursion.